src/sparseml/evaluation/integrations/perplexity.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+import numpy
+import torch
+from datasets import load_dataset
+from torch.nn import CrossEntropyLoss
+from tqdm import tqdm
+
+from huggingface_hub import hf_hub_download
+from sparseml.evaluation.registry import SparseMLEvaluationRegistry
+from sparseml.transformers.integration_helper_functions import create_model
+
+
+@SparseMLEvaluationRegistry.register("perplexity")
+def perplexity_eval(
+    target,
+    datasets: str = "wikitext",
+    batch_size: int = 1,
+    device: Optional[str] = None,
+):
+    nsamples = 100
+    dataset_config_name = infer_dataset_config_name(datasets)
+    task = "text-generation"
+    try:
+        # download the recipe if any
+        recipe_path = hf_hub_download(repo_id=target, filename="recipe.yaml")
+    except Exception:
+        recipe_path = None
+
+    model, other_objects = create_model(
+        source_path=target,
+        device=device,
+        task=task,
+        recipe=recipe_path,
+    )
+
+    tokenizer = other_objects["tokenizer"]
```

Comment:
Could we move the code for dataset processing elsewhere? Or reuse @Satrat 's dataset registries if possible

Reply:
Done!